KRFAnalysisPass/KRF.cpp

```diff
@@ -0,0 +1,296 @@
+#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/DebugInfoMetadata.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/JSON.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/ADT/StringRef.h"
+#include <unordered_set>
+#include <string>
+
+using namespace llvm;
+
+cl::opt<bool> Json ("json", cl::desc("Print output in json format"));
```

Comment:
Prefix this with the pass name to avoid clobbering other `opt` flags: `krf-json` or similar.